packages/core-modules/contracts/BaseProxy.sol

```diff
@@ -0,0 +1,19 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "@synthetixio/core-contracts/contracts/proxy/ForwardingProxy.sol";
+import "./storage/ProxyStorage.sol";
+
+contract BaseProxy is ForwardingProxy, ProxyStorage {
+    constructor(address firstImplementation) {
+        _setImplementation(firstImplementation);
+    }
+
+    function _setImplementation(address newImplementation) internal override {
+        _proxyStorage().implementation = newImplementation;
+    }
+
+    function _getImplementation() internal view override returns (address) {
+        return _proxyStorage().implementation;
+    }
+}
```

Comment:
Also, I had to move the Proxy implementation to `core-modules` because the UpgradeModule uses the ProxyStorage. Soo, the ProxyStorage cannot be located at the main repo.

Reply:
Agree with you @ajsantander but I think it should be **deployable** for testing purposes (maybe the proxy and router used in a test fixture?)

Reply:
Well, you can unit test them? I.e. no integration tests in core-modules.